pkg/controllers/provisioningv2/rke2/provisioningcluster/controller.go

```diff
@@ -282,6 +282,14 @@ func reconcileClusterSpecEtcdRestore(cluster *rancherv1.Cluster, desiredSpec ran
 		changed = true
 		cluster.Spec.DefaultPodSecurityAdmissionConfigurationTemplateName = desiredSpec.DefaultPodSecurityAdmissionConfigurationTemplateName
 	}
+	if !equality.Semantic.DeepEqual(cluster.Spec.ClusterAgentDeploymentCustomization, desiredSpec.ClusterAgentDeploymentCustomization) {
+		changed = true
+		cluster.Spec.ClusterAgentDeploymentCustomization = desiredSpec.ClusterAgentDeploymentCustomization
```

Comment:
@jiaqiluo , same - do we need to worry about DeepCopy here?

Reply:
No, we don't. Because the `desiredSpec` object is not changed anywhere when this function is called.